hibernate-community-dialects/src/main/java/org/hibernate/community/dialect/TimesTenSqlAstTranslator.java

```diff
@@ -143,4 +145,56 @@ protected boolean supportsRowValueConstructorSyntaxInInList() {
 	protected boolean supportsRowValueConstructorSyntaxInQuantifiedPredicates() {
 		return false;
 	}
+
+	@Override
+	protected void renderRowsToClause(Expression offsetClauseExpression, Expression fetchClauseExpression) {
+		// offsetClauseExpression -> firstRow
+		// fetchClauseExpression  -> maxRows
+		final Stack<Clause> clauseStack = getClauseStack();
+
+		if ( offsetClauseExpression == null && fetchClauseExpression != null ) {
+			// We only have a maxRows/limit. We use 'SELECT FIRST n' syntax
+			appendSql("first ");
+			clauseStack.push( Clause.FETCH );
+			try {
+				renderFetchExpression( fetchClauseExpression );
+			}
+			finally {
+				clauseStack.pop();
+			}
+		}
+		else if ( offsetClauseExpression != null && fetchClauseExpression == null ) {
+			throw new UnsupportedOperationException( 
+				"Only passing setFirstResult(m) and not setMaxResults(n) to 'ROWS m TO n' clause not supported." 
+			);
+		}
+		else if ( offsetClauseExpression != null && fetchClauseExpression != null ) {
+			// We have offset and maxRows/limit. We use 'SELECT ROWS offset TO limit' syntax
+			appendSql( "rows " );
+      
+			// Render offset parameter
+			clauseStack.push( Clause.OFFSET );
+			try {
+				renderOffsetExpression( offsetClauseExpression );
+			}
+			finally {
+				clauseStack.pop();
+			}
+
+			appendSql( " to " );
+
+			// Render maxRows/limit parameter
+			clauseStack.push( Clause.FETCH );
+			try {
+				// TimesTen includes both m and n rows of 'ROWS m to n'; 
+				// We need to substract 1 row to fit maxRows
+				renderFetchPlusOffsetExpressionAsLiteral( fetchClauseExpression, offsetClauseExpression, -1 );
```

Comment:
Does TimesTen not support a parameter here?

```suggestion
				renderFetchPlusOffsetExpressionAsSingleParameter( fetchClauseExpression, offsetClauseExpression, -1 );
```